src/pgen/torus.cpp

```diff
@@ -277,6 +251,28 @@ void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
 
         // Radiation
         if (do_rad) {
+          StateDescriptor *opac = pmb->packages.Get("opacity").get();
```

Comment:
This won't work on GPU because you're trying to access host side variables (the `Opacities` object in `Params`) inside a compute kernel which can be on GPU. Instead, you have to pull out some `Opacities` object outside of the compute loop and let Kokkos capture that object by value.

Reply:
Good to know - thanks!

Reply:
:+1: I've been tripped up by this a number of times, and it's something to watch out for especially since we still don't quite have the CI running on gpus.